modules/dop/services/issue/validate.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issue
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/dop/dao"
+)
+
+type issueValidator struct {
+}
+
+type issueValidationConfig struct {
+	iteration *dao.Iteration
+	state     *dao.IssueState
+}
+
+func (v *issueValidator) validateTimeWithInIteration(c *issueValidationConfig, time *time.Time) error {
+	if c == nil {
+		return fmt.Errorf("issue validation config is empty")
+	}
+	if c.iteration == nil {
+		return nil
+	}
+	if !inIterationInterval(c.iteration, time) {
+		return fmt.Errorf("plan finish time is not in the iteration %v interval %v ~ %v",
+			c.iteration.Title, c.iteration.StartedAt.Format("2006-01-02"), c.iteration.FinishedAt.Format("2006-01-02"))
+	}
+	return nil
+}
+
+func inIterationInterval(iteration *dao.Iteration, time *time.Time) bool {
+	if time == nil || iteration == nil {
+		return false
+	}
+	return time.After(*iteration.StartedAt) && time.Before(*iteration.FinishedAt)
```

Comment:
how about the case time == startedAt or item == finishedAt ?

Reply:
changed to inclusive range